cupyx/scipy/sparse/_base.py

```diff
@@ -33,6 +33,23 @@ def __init__(self, maxprint=50):
                 'This class is not intended to be instantiated directly.')
         self.maxprint = maxprint
 
+    def __repr__(self):
+        format_strings = {
+            "coo": "COOrdinate format",
+            "csr": "Compressed Sparse Row format",
+            "csc": "Compressed Sparse Column format",
+            "dia": "DIAgonal format",
+        }
+        matrix_type = str(self.format)
+        stored_elements = self.getnnz() if hasattr(self, "getnnz") \
+            else self.nnz
+        return (
+            f"<{self._shape[0]}x{self._shape[1]} sparse matrix " \
+            f"of type '{self.dtype}'\n " \
+            f"on device {cupy.cuda.get_device_id()} with {stored_elements} "
+            f"stored elements in {format_strings[matrix_type]}>"
+        )
```

Comment:
* Align the output format with SciPy's `__repr__`.
* Device ID should be the device where sparse array resides, not the current device.
* Following the coding convention in CuPy, use single quotes.

```suggestion
        format_name = _format.get(self.format, 'unknown format')
        return (
            f'<{self._shape[0]}x{self._shape[1]} sparse matrix '
            f'of type '{self.dtype.type} '
            f'on device {self.device.id}\n'
            f'\twith {self.nnz} stored elements in {format_name}'
        )
```

Please declare static values at module-level:

```
_format = {
    'coo': 'COOrdinate format',
    'csr': 'Compressed Sparse Row format',
    'csc': 'Compressed Sparse Column format',
    'dia': 'DIAgonal format',
}
```

Reply:
Hi! @kmaehashi Tested this locally, `self.device.id` raises a `NotImplementedError` when calling `self.device`.